torchvision/datasets/folder.py

```diff
@@ -28,11 +28,12 @@ def make_dataset(dir, class_to_idx):
         if not os.path.isdir(d):
             continue
 
-        for filename in os.listdir(d):
-            if is_image_file(filename):
-                path = '{0}/{1}'.format(target, filename)
-                item = (path, class_to_idx[target])
-                images.append(item)
+        for root, _, fnames in sorted(os.walk(d)):
```

Comment:
FWIW, we found it useful to sort them in DIGITS (https://github.com/NVIDIA/DIGITS/pull/296). If you're going to be auto-assigning numeric indices to classnames, it's nice to have the translation be deterministic.